common/iso_config.go

```diff
@@ -23,36 +30,81 @@ func (c *ISOConfig) Prepare(ctx *interpolate.Context) ([]string, []error) {
 	var err error
 	var warnings []string
 
+	if c.RawSingleISOUrl == "" && len(c.ISOUrls) == 0 {
+		errs = append(
+			errs, errors.New("One of iso_url or iso_urls must be specified."))
+	} else if c.RawSingleISOUrl != "" && len(c.ISOUrls) > 0 {
+		errs = append(
+			errs, errors.New("Only one of iso_url or iso_urls may be specified."))
+	} else if c.RawSingleISOUrl != "" {
+		c.ISOUrls = []string{c.RawSingleISOUrl}
+	}
+
 	if c.ISOChecksumType == "" {
 		errs = append(
 			errs, errors.New("The iso_checksum_type must be specified."))
 	} else {
 		c.ISOChecksumType = strings.ToLower(c.ISOChecksumType)
 		if c.ISOChecksumType != "none" {
-			if c.ISOChecksum == "" {
+			if c.ISOChecksum == "" && c.ISOChecksumURL == "" {
 				errs = append(
 					errs, errors.New("Due to large file sizes, an iso_checksum is required"))
+				return warnings, errs
 			} else {
-				c.ISOChecksum = strings.ToLower(c.ISOChecksum)
-			}
+				if h := HashForType(c.ISOChecksumType); h == nil {
+					errs = append(
+						errs, fmt.Errorf("Unsupported checksum type: %s", c.ISOChecksumType))
+					return warnings, errs
+				}
 
-			if h := HashForType(c.ISOChecksumType); h == nil {
-				errs = append(
-					errs,
-					fmt.Errorf("Unsupported checksum type: %s", c.ISOChecksumType))
+				// If iso_checksum has no value use iso_checksum_url instead.
+				if c.ISOChecksum == "" {
+					u, err := url.Parse(c.ISOChecksumURL)
+					if err != nil {
+						errs = append(errs,
+							fmt.Errorf("Error parsing checksum: %s", err))
+						return warnings, errs
+					}
+					switch u.Scheme {
+					case "http", "https", "ftp", "ftps":
```

Comment:
May be drop ftp, ftps ? I don't try this scheme via http.Get and it may not work..